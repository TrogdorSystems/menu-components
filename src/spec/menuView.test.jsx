import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MenuView from '../client/app/menuView';

Enzyme.configure({ adapter: new Adapter() });

describe('MenuView Component', () => {
  test('should render correctly', () => {
    expect(Enzyme.shallow(<MenuView />)).toMatchSnapshot();
  });
});
