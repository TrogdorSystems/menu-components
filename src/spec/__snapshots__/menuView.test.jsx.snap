// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuView Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          <h2>
            Menu
          </h2>
        </div>,
        <div
          className="menuNav"
        >
          <button
            className="lunchBtn"
          >
            Lunch
          </button>
          <button
            className="dinnerBtn"
          >
            Dinner
          </button>
          <button
            className="dessertBtn"
          >
            Dessert
          </button>
          <select
            className="filterBox"
          >
            <option
              value="gluten-free"
            >
              gluten-free
            </option>
            <option
              value="vegan"
            >
              vegan
            </option>
            <option
              value="vegetarian"
            >
              vegetarian
            </option>
            r
          </select>
        </div>,
      ],
      "className": "menuViewBox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            Menu
          </h2>,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Menu",
          },
          "ref": null,
          "rendered": "Menu",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="lunchBtn"
            >
              Lunch
            </button>,
            <button
              className="dinnerBtn"
            >
              Dinner
            </button>,
            <button
              className="dessertBtn"
            >
              Dessert
            </button>,
            <select
              className="filterBox"
            >
              <option
                value="gluten-free"
              >
                gluten-free
              </option>
              <option
                value="vegan"
              >
                vegan
              </option>
              <option
                value="vegetarian"
              >
                vegetarian
              </option>
              r
            </select>,
          ],
          "className": "menuNav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lunch",
              "className": "lunchBtn",
            },
            "ref": null,
            "rendered": "Lunch",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dinner",
              "className": "dinnerBtn",
            },
            "ref": null,
            "rendered": "Dinner",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dessert",
              "className": "dessertBtn",
            },
            "ref": null,
            "rendered": "Dessert",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="gluten-free"
                >
                  gluten-free
                </option>,
                <option
                  value="vegan"
                >
                  vegan
                </option>,
                <option
                  value="vegetarian"
                >
                  vegetarian
                </option>,
                "r",
              ],
              "className": "filterBox",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "gluten-free",
                  "value": "gluten-free",
                },
                "ref": null,
                "rendered": "gluten-free",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "vegan",
                  "value": "vegan",
                },
                "ref": null,
                "rendered": "vegan",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "vegetarian",
                  "value": "vegetarian",
                },
                "ref": null,
                "rendered": "vegetarian",
                "type": "option",
              },
              "r",
            ],
            "type": "select",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            <h2>
              Menu
            </h2>
          </div>,
          <div
            className="menuNav"
          >
            <button
              className="lunchBtn"
            >
              Lunch
            </button>
            <button
              className="dinnerBtn"
            >
              Dinner
            </button>
            <button
              className="dessertBtn"
            >
              Dessert
            </button>
            <select
              className="filterBox"
            >
              <option
                value="gluten-free"
              >
                gluten-free
              </option>
              <option
                value="vegan"
              >
                vegan
              </option>
              <option
                value="vegetarian"
              >
                vegetarian
              </option>
              r
            </select>
          </div>,
        ],
        "className": "menuViewBox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              Menu
            </h2>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Menu",
            },
            "ref": null,
            "rendered": "Menu",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="lunchBtn"
              >
                Lunch
              </button>,
              <button
                className="dinnerBtn"
              >
                Dinner
              </button>,
              <button
                className="dessertBtn"
              >
                Dessert
              </button>,
              <select
                className="filterBox"
              >
                <option
                  value="gluten-free"
                >
                  gluten-free
                </option>
                <option
                  value="vegan"
                >
                  vegan
                </option>
                <option
                  value="vegetarian"
                >
                  vegetarian
                </option>
                r
              </select>,
            ],
            "className": "menuNav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lunch",
                "className": "lunchBtn",
              },
              "ref": null,
              "rendered": "Lunch",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dinner",
                "className": "dinnerBtn",
              },
              "ref": null,
              "rendered": "Dinner",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dessert",
                "className": "dessertBtn",
              },
              "ref": null,
              "rendered": "Dessert",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="gluten-free"
                  >
                    gluten-free
                  </option>,
                  <option
                    value="vegan"
                  >
                    vegan
                  </option>,
                  <option
                    value="vegetarian"
                  >
                    vegetarian
                  </option>,
                  "r",
                ],
                "className": "filterBox",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "gluten-free",
                    "value": "gluten-free",
                  },
                  "ref": null,
                  "rendered": "gluten-free",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "vegan",
                    "value": "vegan",
                  },
                  "ref": null,
                  "rendered": "vegan",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "vegetarian",
                    "value": "vegetarian",
                  },
                  "ref": null,
                  "rendered": "vegetarian",
                  "type": "option",
                },
                "r",
              ],
              "type": "select",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
